deno_dist/helper/ssg/index.ts

```diff
@@ -68,19 +68,21 @@ interface SSGParam {
   [key: string]: string
 }
 type SSGParams = SSGParam[]
-interface SSGParamsMiddleware {
-  (generateParams: (c: Context) => SSGParams | Promise<SSGParams>): MiddlewareHandler
-}
+
 type AddedSSGDataRequest = Request & {
   ssgParams?: SSGParams
 }
 /**
  * Define SSG Route
  */
-export const ssgParams: SSGParamsMiddleware = (generateParams) => async (c, next) => {
```

Comment:
You can write them here:

```ts
interface SSGParamsMiddleware {
  <E extends Env = Env>(
    generateParams: (c: Context<E>) => SSGParams | Promise<SSGParams>
  ): MiddlewareHandler<E>
}
```

I would suggest separating implementation and interface in this case.